contrib/zpages/src/main/java/io/opencensus/contrib/zpages/RpczZPageHandler.java

```diff
@@ -0,0 +1,472 @@
+    } catch (Throwable t) {
+      out.write("Errors while generate the HTML page " + t);
+    }
+    out.write("</body>\n");
+    out.write("</html>\n");
+    out.close();
+  }
+
+  private void emitHtmlBody(PrintWriter out) {
+    Formatter formatter = new Formatter(out, Locale.US);
+    out.write("<h1>RPC Stats</h1>");
+    out.write("<p></p>");
+    emitSummaryTable(out, formatter, /* isReceived= */ false);
+    emitSummaryTable(out, formatter, /* isReceived= */ true);
+  }
+
+  private void emitSummaryTable(PrintWriter out, Formatter formatter, boolean isReceived) {
+    formatter.format(
+        "<p><table bgcolor=%s width=100%%><tr align=center><td><font size=+2>"
+            + "%s</font></td></tr></table></p>",
+        TITLE_COLOR, (isReceived ? RECEIVED : SENT));
+    formatter.format(
+        "<table bgcolor=%s frame=box cellspacing=0 cellpadding=2>", TABLE_HEADER_COLOR);
+    emitSummaryTableHeader(out, formatter);
+    Map<String, StatsSnapshot> snapshots = getStatsSnapshots(isReceived);
+    for (Entry<String, StatsSnapshot> entry : snapshots.entrySet()) {
+      emitSummaryTableRows(out, formatter, entry.getValue(), entry.getKey());
+    }
+    out.write("</table>");
+  }
+
+  private static void emitSummaryTableHeader(PrintWriter out, Formatter formatter) {
+    // First line.
+    formatter.format("<tr bgcolor=%s>", TABLE_ROW_COLOR);
+    out.write("<th></th><td></td>");
+    for (String rpcStatsType : RPC_STATS_TYPES) {
+      formatter.format("<th class=\"l1\" colspan=3>%s</th><td></td>", rpcStatsType);
+    }
+    out.write("</tr>");
+
+    // Second line.
+    formatter.format("<tr bgcolor=%s>", TABLE_ROW_COLOR);
+    out.write("<th align=left>Method</th><td>&nbsp;&nbsp;&nbsp;&nbsp;</td>");
+    for (int i = 0; i < RPC_STATS_TYPES.size(); i++) {
+      out.write("<th align=right>Min.</th><th align=right>Hr.</th><th align=right>Tot.</th>");
+      if (i != RPC_STATS_TYPES.size() - 1) { // Add spaces between each column.
+        out.write("<td>&nbsp;&nbsp;&nbsp;&nbsp;</td>");
+      }
+    }
+    out.write("</tr>");
+
+    // Empty line.
+    out.write("<tr><td colspan=33><font size=-2>&nbsp;</font></td></tr>");
+  }
+
+  private static void emitSummaryTableRows(
+      PrintWriter out, Formatter formatter, StatsSnapshot snapshot, String method) {
+    out.write("<tr>");
+    formatter.format("<td><b>%s</b></td>", method);
+    out.write("<td></td>");
+    formatter.format("<td align=\"right\">%d</td>", snapshot.countLastMinute);
+    formatter.format("<td align=\"right\">%d</td>", snapshot.countLastHour);
+    formatter.format("<td align=\"right\">%d</td><td></td>", snapshot.countTotal);
+    formatter.format("<td align=\"right\">%.3f</td>", snapshot.avgLatencyLastMinute);
+    formatter.format("<td align=\"right\">%.3f</td>", snapshot.avgLatencyLastHour);
+    formatter.format("<td align=\"right\">%.3f</td><td></td>", snapshot.avgLatencyTotal);
+    // We don't record max/min values for minute/hour views in Java.
```

Comment:
Probably we should not have this column with max latency. We can add later a percentile column if needed.

Reply:
Saw your comments under https://github.com/census-instrumentation/opencensus-specs/issues/47, I've removed max latency for the moment and left a TODO to add percentiles support.